compiler/rustc_target/src/abi/mod.rs

```diff
@@ -389,6 +389,21 @@ impl Size {
         // Truncate (shift left to drop out leftover values, shift right to fill with zeroes).
         (value << shift) >> shift
     }
+
+    #[inline]
+    pub fn signed_min(&self) -> i128 {
+        self.sign_extend(1_u128 << (self.bits() - 1)) as i128
+    }
+
+    #[inline]
+    pub fn signed_max(&self) -> i128 {
+        i128::MAX >> (128 - self.bits())
+    }
+
+    #[inline]
+    pub fn unsigned_max(&self) -> u128 {
+        u128::MAX >> (128 - self.bits())
+    }
```

Comment:
Well, we already have `sign_extend` and `truncate` on `Size`... and I think it would be useful to have these min/max methods somewhere as this is annoying bitfiddling.

`Integer` would be annoying unless there is an easy way to go from a `Size` to an integer. We will typically have the size of the data we are working on available as a `Size`.

Reply:
Since lack of good alternatives(at least discussed), I moved them back into `Size` again.

Reply:
In that case, can they say `(un)signed_int` instead of just `(un)signed`? (or `uint` and `sint`)

I still don't like them on `Size` but at least I'm less opposed to them that way, I guess.